code/modules/vehicles/_hitbox.dm

```diff
@@ -230,3 +248,135 @@
 	if(canstep)
 		return NONE
 	return COMPONENT_DRIVER_BLOCK_MOVE
+
+//3x4
+/obj/hitbox/rectangle
+
+	bound_x = -32
+	bound_y = -64
+	vehicle_length = 128
+	vehicle_width = 96
+
+/obj/hitbox/rectangle/owner_turned(datum/source, old_dir, new_dir)
+	. = ..()
+	if(!.)
+		return
+	switch(new_dir)
+		if(NORTH)
+			bound_height = vehicle_length
+			bound_width = vehicle_width
+			bound_x = -32
+			bound_y = -32
+			root.pixel_x = -65
+			root.pixel_y = -48
+		if(SOUTH)
+			bound_height = vehicle_length
+			bound_width = vehicle_width
+			bound_x = -32
+			bound_y = -64
+			root.pixel_x = -65
+			root.pixel_y = -80
+		if(WEST)
+			bound_height = vehicle_width
+			bound_width = vehicle_length
+			bound_x = -64
+			bound_y = -32
+			root.pixel_x = -80
+			root.pixel_y = -56
+		if(EAST)
+			bound_height = vehicle_width
+			bound_width = vehicle_length
+			bound_x = -32
+			bound_y = -32
+			root.pixel_x = -48
+			root.pixel_y = -56
+
+	var/angle_change = dir2angle(new_dir) - dir2angle(old_dir)
+	//north needing to be considered 0 OR 360 is inconvenient, I'm sure there is a non ungabrain way to do this
+	switch(angle_change)
+		if(-270)
+			angle_change = 90
+		if(270)
+			angle_change = -90
+	for(var/mob/living/desant AS in tank_desants)
+		if(desant.loc == root.loc)
+			continue
+		var/new_x
+		var/new_y
+		if(angle_change > 0) //clockwise turn
+			new_x = root.x + (desant.y - root.y)
+			new_y = root.y - (desant.x - root.x)
+		else //anti-clockwise
+			new_x = root.x - (desant.y - root.y)
+			new_y = root.y + (desant.x - root.x)
+
+		desant.forceMove(locate(new_x, new_y, z))
+
+/obj/hitbox/rectangle/on_attempt_drive(atom/movable/movable_parent, mob/living/user, direction) //haven't baked in strafe into the base type since its only used by the SOM tank currently
+	if(ISDIAGONALDIR(direction))
+		return COMPONENT_DRIVER_BLOCK_MOVE
```

Comment:
just bake it in now lol? idk why we wouldnt have it there

Reply:
because lazy. Updated now.

Reply:
not updated?

Reply:
its updated, deleted the comment now.